### ISSUE TITLE
Adding context information to events (LoadContextsIntegration)

### ISSUE THREAD

**Issue by kuhnroyal** [getsentry/sentry-dart#224]
This might not be a full feature request maybe more of a documentation request, not sure. It is about Flutter integration.

With the 3.x version I have manually built `SentryEvent`s and added all the context information like `Device`, `App`, `OperatingSystem` and so on, using a bunch of different plugins (connectivity, timezone, platform etc.). For me it is not very clear which of this information is now automatically added due to the native integrations. So there are a couple questions that I have not found (fully) answered in code or documentation.

1. Which device information does the `LoadContextsIntegration` produce?
2. Is it planned to support `LoadContextsIntegration` on Android?
3. When there are multiple `EventProcessor`s it is currently hard to add information to the nested structure of the event. Only `SentryEvent` has a `copyWith(...)` function. Ideally there could be similar copy functions for all the nested context classes so that one could do things like the following which is currently very hard to achieve as all the children have to be rebuild completely.

```dart
final connectivityResult = await Connectivity().checkConnectivity();
final online = connectivityResult != ConnectivityResult.none;

return event.copyWith(
  contexts: event.contexts.copyWith(
    device: (event.contexts.device ?? Device()).copyWith(
      online: online,
    ),
  ),
);
```

**Comment by marandaneto** [comment]
@kuhnroyal thanks for raising this.

`LoadContextsIntegration` is an iOS only integration, which is gotten from the Cocoa SDK before building the event, The Contexts include App, Device, OS, etc...

The Android SDK enriches the event with the device context afterward, once it's received via the message channel, it's a different way of working and we intend to unify this behavior.

True, ideally all the data classes should have the `copyWith`, as it's the easiest way to mutate an event, I'll tag it as an enhancement.

**Comment by marandaneto** [comment]
btw pay attention that using the new Unified API, Contexts would be ideally added via our Scope

https://docs.sentry.io/platforms/dart/enriching-events/context/

so you set it only once, and those values are merged into every captured event.
of course, this would make sense for only static values that don't change.

**Comment by marandaneto** [comment]
it's gonna be on the next version :)
